### PR TITLE
Remove accumulate_grad version_counter check.

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

**Comment by gchanan** [comment]
I ran into this while trying to add some scalar tests to autograd; essentially because we have scalars in Variables (via ATen), but not with tensors, I need to construct my inputs using only the Variable API (i.e. not escaping to Tensor API).  So what is currently something like:
```
grad_y = (Variable(torch.randn(dummy_out.size()), requires_grad=dummy_out.requires_grad),)
```
becomes something like:
```
grad_y = dummy_out.new(dummy_out.size())
grad_y.normal_()
grad_y.requires_grad=True
```

but this fails in the backward in the accumulate_grad check that I've removed.  Given that we are planing to merge the Variable and Tensor API, it seems reasonable to have a construct like this.

**Comment by colesbury** [comment]
Yeah, I don't see this check doing anything useful anymore.

**Comment by gchanan** [comment]
I've been trying to figure out what this check is meant to prevent, and I think here's a case:
```
 a = Variable(torch.randn(2, 3))
b = a[:]
b.requires_grad = True
a.add_(5)
with self.assertRaises(RuntimeError):
    b.add_(5)
b.sum().backward()
```

This is caused by the fact that views require grad if the base does, but not visa-versa.

**Comment by gchanan** [comment]
Do we need to make it so that requires_grad on views implies requires_grad on the base before we do this?

**Comment by gchanan** [comment]
I'm not really sure if the above is an issue or not; it does seem strange that one can modify the Variable via a, but not b, but maybe that's a separate issue.

File: test/test_autograd.py [hunk 1]
```diff
@@ -759,6 +759,31 @@ def test_requires_grad_inplace(self):
         a += b
         self.assertTrue(a.requires_grad)
 
+    def test_no_requires_grad_inplace(self):
+        # basic case, shoudl be able to modify inplace while requires_grad is False
```
Comment (colesbury) [inline test/test_autograd.py]: (sp)

**Review by colesbury — APPROVED** [review]

**Comment by apaszke** [comment]
@gchanan IIRC it should already be the case that `requires_grad` on views implies `requires_grad` on the base. Have you ever seen it not work that way?

**Comment by gchanan** [comment]
it doesn't work in that direction, see:
```
inline bool Variable::requires_grad() const {
  return get()->_requires_grad || get()->_grad_fn || (is_view() && base().requires_grad());
}
```

@colesbury said it's to handle cases in python functions when an input is returned from the forward; we consider that a view (and set requires_grad on it), but don't want to affect the base.

**Comment by apaszke** [comment]
Ugh, this is so weird. We should have a single document that describes all of these semantics.

**Comment by gchanan** [comment]
failures look CI related.

**Pull request merged**